javascript/elements/updates_for_element.js

```diff
@@ -40,12 +40,15 @@ export default class UpdatesForElement extends SubscribingElement {
     }
   }
 
-  async update () {
+  async update (data) {
     const identifier = this.getAttribute('identifier')
     const query = `updates-for[identifier="${identifier}"]`
     const blocks = document.querySelectorAll(query)
     if (blocks[0] !== this) return
 
+    const only = this.getAttribute('only')
+    if (only && data.changed && !data.changed.includes(only)) return
```

Comment:
@leastbad just out of interest, would we be able to tweak this to allow for passing multiple fields into the `only` option?

I was thinking of something along the lines of:

```js
const only = this.getAttribute('only')?.split(" ")
if (only && data.changed && !only.some(attribute => data.changed.includes(attribute))) return
```

Which, if I understand correctly, would allow you to do:

```erb
<%= updates_for User.first, only: [:name, :email, :favorite_abba_song] do %>
  <p><%= rand(1..1000) %></p>
<% end %>
```

Reply:
`some` would make it an `OR`, right?

Just to play the devil's advocate, can you make a strong case why we'd go with `OR` over `AND` (or `NOR`/`XOR` 😉)

Reply:
BTW, this isn't just a pedantic question... technically, `AND` would allow developers to build up simple logical constraints eg "this one needs name and age, that one needs name and gender"

Whereas `OR` is probably better for an admin scenario where you have a bunch of user-editable data alongside a busy attribute such as a counter cache that you seek to filter out.

Reply:
What do you think, @julianrubisch?

Reply:
Just reading up on it here...

Very valid question. For me it boils down to the principle of least surprise:

If in Rails I do `before_action only: %i[new edit]` I expect it to work on both, right? So that'd make a case for an `OR`.

Can anyone think of a counter-example?

Reply:
@leastbad yes, `some` would make it an `OR`.

I hadn't really considered the case for `AND` to be honest. And this is probably just a lack of imagination on my part, but I can't picture an obvious application for it.

I was operating on the assumption that you'd usually want `only` to include all attributes that feature within the `updates_for` block.

```erb
<%= updates_for User.first, only: [:name, :age] do |user| %>
  <p><%= user.name %></p>
  <p><%= user.age %></p>
<% end %>

<%= updates_for User.first, only: [:name, :gender] do |user| %>
  <p><%= user.name %></p>
  <p><%= user.gender %></p>
<% end %>
```

In cases like this, I imagine you'd usually want to update the contents of the block any time _any_ supplied attribute is updated, to keep the HTML content in-sync with the database.

I'm sure there are some cases in which `AND` would be the better fit, I just have the sense they'd be a lot more rare. Also interested in @julianrubisch's thoughts on this :)

Reply:
I think both of you made the right points.

@andrewerlanger, do you feel up to making a PR against this PR to implement the changes to both `updates_for_element.js` and `cable_ready_helper.rb`?

It's important that the helper accept a string, a symbol and an array of strings or symbols, and of course, the web component will have to be able to handle whatever shows up.

I actually don't know how `tag` will serialize an array. Will it do space-separated tokens? comma-separated?

Reply:
@leastbad happy to take a stab at it!

I'm not sure we'll actually need to make any changes to `cable_ready_helper.rb` though. As far as I can tell, any array we pass to `only` ends up becoming a (space-separated) string by the time it makes its way to the javascript.

I just added this to the `updates_for_element.js` code...

```js
const only = this.getAttribute('only')?.split(" ")
console.log(only)
```

...and got the following:
```js
// only: [:name, :age]
['name', 'age']

// only: ["name", "age"]
['name', 'age']

// only: "name age"
['name', 'age']

// only: "name"
['name']

// only: :name
['name']
```

So I think we should be in the clear on that front, but let me know if I've missed something.

I'll take a shot at those `updates_for_element.js` changes soon :)

Reply:
Nice work! I had to make a quick tweak, but it's now in there and seems to be working great.